src/doc_builder/commands/commit.py

```diff
@@ -0,0 +1,145 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import base64
+import re
+from pathlib import Path
+
+import requests
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+from gql.transport.exceptions import TransportQueryError
+
+
+_re_head_oid = re.compile(r"\"sha\":\"([0-9a-z]+)\"")
+
+
+def get_head_oid(repo_id, token, branch="main"):
+    """
+    Returns last commit sha from repostory `repo_id` & branch `branch`
+    """
+    res = requests.get(
+        f"https://api.github.com/repos/{repo_id}/git/refs/heads/${branch}",
+        headers={"Authorization": f"bearer {token}"},
+    )
+    if res.status_code == 200:
+        head_oid = _re_head_oid.search(res.text)[1]
+        return head_oid
+    else:
+        raise Exception("get_head_oid failed: {{status: {res.status}, msg:{res.text}}}")
+
+
+def create_additions_str(path_to_built_docs):
+    """
+    Given `path_to_built_docs` dir, returns a str that is in format: `{path: "some_path", contents: "base64_repr_contents"}, ...`
+    see more here: https://docs.github.com/en/graphql/reference/input-objects#filechanges
+    """
+    p = Path(path_to_built_docs)
+    files = [x for x in p.glob("**/*") if x.is_file()]
+    additions = []
+
+    for fpath in files:
+        with open(fpath, "rb") as reader:
+            content = reader.read()
+            content_base64 = base64.b64encode(content)
+            content_base64 = str(content_base64, "utf-8")
+            additions.append(f'{{path: "{fpath}", contents: "{content_base64}"}}')
+
+    additions_str = ", ".join(additions)
+    return additions_str
+
+
+def commit_additions(additions_str, repo_id, head_oid, token, commit_msg):
+    """
+    Commits additions to a repository using Github GraphQL mutation `createCommitOnBranch`
+    see more here: https://docs.github.com/en/graphql/reference/mutations#createcommitonbranch
+    """
+    # Select your transport with a defined url endpoint
+    transport = AIOHTTPTransport(url="https://api.github.com/graphql", headers={"Authorization": f"bearer {token}"})
+    # Create a GraphQL client using the defined transport
+    client = Client(transport=transport, fetch_schema_from_transport=True)
+    # Provide a GraphQL query
+    query = gql(
+        f"""
+  mutation{{
+    createCommitOnBranch(
+      input: {{
+        branch:{{
+          repositoryNameWithOwner: "{repo_id}",
+          branchName: "main"
+        }},
+        message: {{
+          headline: "{commit_msg}"
+        }},
+        fileChanges: {{
+          additions: [{additions_str}]
+        }},
+        expectedHeadOid: "{head_oid}"
+      }}
+    ) {{
+      clientMutationId
+    }}
+  }}
+  """
+    )
+    # Execute the query on the transport
+    result = client.execute(query)
```

Comment:
might be better to use graphql "variables" and not create the string dynamically

Reply:
(sorry i know this is not ready for review but I was curious to take a look 😅)

Reply:
handled in https://github.com/huggingface/doc-builder/pull/214/commits/a3ffc278896381771a5a923c7db362de29dd24be
using "variables" actually make the implemntation cleaner, great suggestion!